agent/engine/docker_task_engine_test.go

```diff
@@ -1722,6 +1722,43 @@ func TestPullAndUpdateContainerReference(t *testing.T) {
 	assert.Equal(t, dockerapi.DockerContainerMetadata{}, metadata, "expected empty metadata")
 }
 
+// TestPullAndUpdateContainerReferenceWithCachedImage checks whether a container is added to task engine state when
+// the container is an essential container, the image is cached and DependentContainersPullUpfront is enabled.
+func TestPullAndUpdateContainerReferenceWithCachedImage(t *testing.T) {
```

Comment:
could you add a test case for the negative scenario? - image pull behavior is always

Reply:
The negative test case is added, named `TestPullAndUpdateContainerReferenceFailedToGetImage` [here](https://github.com/aws/amazon-ecs-agent/pull/2800/files#diff-aa8aba86b2dbb7c182fa55d44aed378f40e350d5ae11405e017dd56d6b58ad04R1765-R1798) 
Thank you!